### PR TITLE
Add internet gateways

### LINKED ISSUE #28: [aws] Add Internet Gateway code
Add code to interface with an internet gateway

### PR DESCRIPTION
**Pull request by vincepri**
Closes #28 
Signed-off-by: Vince Prignano <vince@vincepri.com>

### CODE REVIEW

File: cloud/aws/providerconfig/v1alpha1/types.go [hunk 1]
```diff
@@ -160,6 +160,9 @@ type Network struct {
 	// VPC defines the cluster vpc.
 	VPC *VPC `json:"vpc"`
 
+	// InternetGatewayID is the id of the internet gateway associated with the VPC.
+	InternetGatewayID *string `json:"internetGatewayId"`
```
Comment (chuckha) [inline cloud/aws/providerconfig/v1alpha1/types.go]: i think you have to regen the api stuff, `make generate`

Comment (vincepri) [reply cloud/aws/providerconfig/v1alpha1/types.go]: Good point, will do

Comment (vincepri) [reply cloud/aws/providerconfig/v1alpha1/types.go]: Done!

**Pull request merged**